pom.xml

```diff
@@ -415,6 +415,7 @@
             <groupId>org.slf4j</groupId>
             <artifactId>slf4j-api</artifactId>
             <version>1.7.36</version>
+            <scope>compile</scope>
```

Comment:
`compile` is already the default scope for a dependency with Maven. Adding it doesn't change anything : https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html#dependency-scope

I guess if we want to make it optional it should not be automatically added transitively as a dependency and we should use the `<optional>true</optional>` configuration ? https://maven.apache.org/guides/introduction/introduction-to-optional-and-excludes-dependencies.html

Reply:
i am not very familiar with Maven, but when we reference the SLF4J package, java needs to know the package, even if it is not included in the build, to compile the code.

i think we can leave it as is, and leave slf4j in the makefile to compile NativeDB.java

Reply:
The `<optional>` configuration does not exclude the dependency for the `sqlite-jdbc` build. It will still be added to the classpath when compiling.

The `<optional>` configuration will only affect builds which depends on `sqlite-jdbc`. The `slf4j-api` will not be added as a transitive dependency. It should make the consumption easier like:
 - one that don't want to use SLF4J would add:
```
<dependencies>
    <dependency>
      <groupId>org.xerial</groupId>
      <artifactId>sqlite-jdbc</artifactId>
      <version>3.46.1.0</version>
    </dependency>
</dependencies>
```
instead of:
```
<dependencies>
    <dependency>
      <groupId>org.xerial</groupId>
      <artifactId>sqlite-jdbc</artifactId>
      <version>3.46.1.0</version>
    </dependency>
    <exclusions>
       <exclusion>
         <groupId>org.slf4j</groupId>
         <artifactId>slf4j-api</artifactId>
       </exclusion>
     </exclusions>
</dependencies>
```
 - one that want to use SLF4J will add:
```
<dependencies>
    <dependency>
      <groupId>org.xerial</groupId>
      <artifactId>sqlite-jdbc</artifactId>
      <version>3.46.1.0</version>
    </dependency>
<dependency>
      <groupId>org.slf4j</groupId>
      <artifactId>slf4j-api</artifactId>
      <version>1.7.36</version>
    </dependency>
</dependencies>
```
But if we want to use SLF4J the dependency should be already declared somewhere 😇.

It just triggers me because we say SLF4J is optional but it will still be added in the dependency graph if we don't exclude it explicitly. It may be not what people expect when using `sqlite-jdbc` but this is not necessarily a big deal if this is documented 😇.

Reply:
i guess optional would be better than compile then ?

Reply:
In fact this is an addition to the `compile` scope, it would be: 

```
<dependency>
            <groupId>org.slf4j</groupId>
            <artifactId>slf4j-api</artifactId>
            <version>1.7.36</version>
            <scope>compile</scope>
            <optional>true</optional>
</dependency>
```

or just (which is equivalent):

```
<dependency>
            <groupId>org.slf4j</groupId>
            <artifactId>slf4j-api</artifactId>
            <version>1.7.36</version>
            <optional>true</optional>
</dependency>
```

Because with Maven, when a scope is not set, it is `compile` by default.

The other way to reproduce the same transitive mechanism behavior (not adding slf4j-api automatically in the classpath when you depend on sqlite-jdbc) would be to replace the `compile` scope by `provided`. But it does change a little the meaning because it would indicate that we really need `slf4j-api` at runtime and that it should be provided by something else (an application server, etc): https://stackoverflow.com/questions/40393098/when-to-use-optional-dependencies-and-when-to-use-provided-scope

My 2 cents is to make the dependency optional in `sqlite-jdbc` which will not resolve `slf4j-api` transitively when we depend on `sqlite-jdbc` and let the consumer use this logging facade or not 😇.

Reply:
> My 2 cents is to make the dependency `<optional>` in `sqlite-jdbc` which will not resolve `slf4j-api` transitively when we depend on `sqlite-jdbc` and let the consumer use this logging facade or not 😇.

i am OK with that

Reply:
> My 2 cents is to make the dependency <optional>

I did not know this! I will make the change and re-push.